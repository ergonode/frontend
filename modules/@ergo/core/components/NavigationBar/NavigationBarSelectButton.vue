/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton
        :is-selected="isFocused"
        @click.native="onClick">
        <slot name="input" />
        <FadeTransition>
            <div
                v-if="isFocused"
                ref="menu"
                class="dropdown">
                <slot name="dropdown" />
            </div>
        </FadeTransition>
    </NavigationBarButton>
</template>

<script>
import {
    isMouseOutOfBoundsElement,
} from '@Core/models/drag_and_drop/helpers';

export default {
    name: 'NavigationBarSelectButton',
    components: {
        NavigationBarButton: () => import('@Core/components/NavigationBar/NavigationBarButton'),
        FadeTransition: () => import('@Core/components/Transitions/FadeTransition'),
    },
    data() {
        return {
            isFocused: false,
            isClickedOutside: false,
        };
    },
    beforeDestroy() {
        window.removeEventListener('click', this.onClickOutside);
    },
    methods: {
        onClick() {
            if (!this.isFocused) {
                window.addEventListener('click', this.onClickOutside);
            }
        },
        onClickOutside(event) {
            let isClickedInsideMenu = false;

            if (this.isFocused) {
                const { pageX, pageY } = event;
                const { menu } = this.$refs;

                isClickedInsideMenu = !isMouseOutOfBoundsElement(menu, pageX, pageY);

                if (!isClickedInsideMenu) {
                    this.isFocused = false;
                }
            } else {
                this.isFocused = !this.isFocused;
            }

            if (!this.isFocused) {
                window.removeEventListener('click', this.onClickOutside);
            }

            this.$emit('focus', this.isFocused);
        },
    },
};
</script>

<style lang="scss" scoped>
    .dropdown {
        position: absolute;
        top: 48px;
        right: 0;
        z-index: $Z_INDEX_NAV;
        display: flex;
        background-color: $WHITE;
        box-shadow: $ELEVATOR_2_DP;
        transition: all 0.4s ease;
        color: $GRAPHITE;
        cursor: default;
    }
</style>
