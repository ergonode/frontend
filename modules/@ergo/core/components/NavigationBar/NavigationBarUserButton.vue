/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarSelectButton
        data-cy="navBarDropDown"
        @focus="onFocus">
        <template #input>
            <UserAvatar
                :image-id="user.avatar_id"
                :name="user.first_name" />
            <span
                class="title"
                v-text="capitalizedUserFirstName || capitalizedUserLastName" />
            <div class="icon-wrapper">
                <IconArrowDropDown
                    :fill-color="whiteColor"
                    :state="arrowIconState" />
            </div>
        </template>
        <template #dropdown>
            <NavigationBarUserSelectContent
                :avatar-id="user.avatar_id"
                :initials="userInitials"
                :email="user.email" />
        </template>
    </NavigationBarSelectButton>
</template>

<script>
import { mapState } from 'vuex';
import { WHITE } from '@Core/assets/scss/_js-variables/colors.scss';
import { toCapitalize } from '@Core/models/stringWrapper';
import { ARROW } from '@Core/defaults/icons';

export default {
    name: 'NavigationBarUserButton',
    components: {
        NavigationBarSelectButton: () => import('@Core/components/NavigationBar/NavigationBarSelectButton'),
        NavigationBarUserSelectContent: () => import('@Core/components/NavigationBar/NavigationBarUserSelectContent'),
        IconArrowDropDown: () => import('@Core/components/Icons/Arrows/IconArrowDropDown'),
        UserAvatar: () => import('@Core/components/Multimedia/UserAvatar'),
    },
    data() {
        return {
            isFocused: false,
        };
    },
    computed: {
        ...mapState('authentication', {
            user: state => state.user,
        }),
        whiteColor() {
            return WHITE;
        },
        arrowIconState() {
            return this.isFocused ? ARROW.UP : ARROW.DOWN;
        },
        capitalizedUserFirstName() {
            const { first_name: firstName } = this.user;

            return toCapitalize(firstName);
        },
        capitalizedUserLastName() {
            const { last_name: lastName } = this.user;

            return toCapitalize(lastName);
        },
        userInitials() {
            return `${this.capitalizedUserFirstName} ${this.capitalizedUserLastName}`;
        },
    },
    methods: {
        onFocus(isFocused) {
            this.isFocused = isFocused;
        },
    },
};
</script>

<style lang="scss" scoped>
    .icon-wrapper {
        display: flex;
        margin-right: -8px;
    }

    .title {
        color: $WHITE;
        font: $FONT_SEMI_BOLD_14_20;
        margin-left: 8px;
    }
</style>
