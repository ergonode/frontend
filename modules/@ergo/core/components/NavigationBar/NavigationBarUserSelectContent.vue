/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div
        data-cy="navBarContent"
        class="user-content">
        <NavigationBarUserSelectContentHeader
            :initials="initials"
            :avatar-id="avatarId"
            :email="email" />
        <Divider />
        <MenuList
            v-for="(item, index) in userMenuElements"
            :key="index"
            :section-title="item.title"
            :section-menu="item.menu" />
        <div class="footer">
            <Button
                data-cy="logoutButton"
                :theme="secondaryTheme"
                title="LOG OUT"
                @click.native="onLogout" />
        </div>
    </div>
</template>

<script>
import { mapActions } from 'vuex';
import { THEME } from '@Core/defaults/theme';
import navigationBarUserMenu from '@Core/models/navigation/navigationBarUserMenu';

export default {
    name: 'NavigationBarUserSelectContent',
    components: {
        Divider: () => import('@Core/components/Dividers/Divider'),
        NavigationBarUserSelectContentHeader: () => import('@Core/components/NavigationBar/NavigationBarUserSelectContentHeader'),
        MenuList: () => import('@Core/components/MenuList/MenuList'),
        Button: () => import('@Core/components/Buttons/Button'),
    },
    props: {
        avatarId: {
            type: String,
            default: '',
        },
        initials: {
            type: String,
            required: true,
        },
        email: {
            type: String,
            required: true,
        },
    },
    data() {
        return {
            userMenuElements: navigationBarUserMenu,
        };
    },
    computed: {
        secondaryTheme() {
            return THEME.SECONDARY;
        },
    },
    methods: {
        ...mapActions('authentication', [
            'setLoggedState',
        ]),
        onLogout() {
            this.setLoggedState(false);
            this.$router.push({ name: 'index' });
        },
    },
};
</script>

<style lang="scss" scoped>
    .user-content {
        display: flex;
        flex-direction: column;
        width: 320px;

        .footer {
            display: flex;
            justify-content: flex-end;
            padding: 16px;
        }
    }
</style>
