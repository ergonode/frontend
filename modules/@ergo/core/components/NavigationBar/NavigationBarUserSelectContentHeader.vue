/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="header">
        <UserAvatar
            :image-id="avatarId"
            :name="initials"
            :avatar-size="64" />
        <span
            class="header__initials"
            v-text="initials" />
        <span
            class="header__email"
            v-text="email" />
    </div>
</template>

<script>
import { GRAPHITE } from '@Core/assets/scss/_js-variables/colors.scss';

export default {
    name: 'NavigationBarUserSelectContentHeader',
    components: {
        UserAvatar: () => import('@Core/components/Multimedia/UserAvatar'),
    },
    props: {
        initials: {
            type: String,
            required: true,
        },
        email: {
            type: String,
            required: true,
        },
        avatarId: {
            type: String,
            default: '',
        },
    },
    computed: {
        graphiteColor() {
            return GRAPHITE;
        },
    },
};
</script>

<style lang="scss" scoped>
    .header {
        display: grid;
        grid-template-columns: 72px auto;
        grid-column-gap: 24px;
        align-items: center;
        margin: 24px;

        &__initials, &__email {
            grid-column: 2;
            text-align: left;
        }

        &__initials {
            grid-row: 1;
            align-self: flex-end;
            color: $GRAPHITE_DARK;
            letter-spacing: 0.1px;
            font: $FONT_MEDIUM_16_24;
        }

        &__email {
            grid-row: 2;
            align-self: flex-start;
            color: $GREY_DARK;
            font: $FONT_MEDIUM_12_16;
        }

        .user-avatar {
            grid-row: 1 / 3;
        }
    }
</style>
