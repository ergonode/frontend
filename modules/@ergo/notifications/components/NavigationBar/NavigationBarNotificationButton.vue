/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarSelectButton
        @focus="onFocus">
        <template #input>
            <IconBell :fill-color="whiteColor" />
            <NotificationBadge
                v-if="notificationsCount > 0"
                :number="notificationsCount" />
        </template>
        <template #dropdown>
            <div class="notifications-dropdown">
                <List v-if="notifications.length">
                    <NotificationsListElement
                        v-for="notification in notifications"
                        :key="notification.id"
                        :notification="notification" />
                    <NotificationsListFooter>
                        <Button
                            title="SEE ALL NOTIFICATIONS"
                            @click.native="navigateToAllNotifications" />
                    </NotificationsListFooter>
                </List>
                <ListPlaceholder
                    v-else
                    title="Nothing to see here"
                    subtitle="Here you can see important notifications of product update"
                    :bg-url="require('@Core/assets/images/placeholders/notify.svg')" />
            </div>
        </template>
    </NavigationBarSelectButton>
</template>

<script>
import { mapState, mapActions } from 'vuex';
import { WHITE } from '@Core/assets/scss/_js-variables/colors.scss';

export default {
    name: 'NavigationBarNotificationButton',
    components: {
        NavigationBarSelectButton: () => import('@Core/components/NavigationBar/NavigationBarSelectButton'),
        IconBell: () => import('@Core/components/Icons/Menu/IconBell'),
        NotificationBadge: () => import('@Core/components/Badges/NotificationBadge'),
        Button: () => import('@Core/components/Buttons/Button'),
        List: () => import('@Core/components/List/List'),
        NotificationsListElement: () => import('@Notifications/components/List/NotificationsListElement'),
        NotificationsListFooter: () => import('@Notifications/components/List/NotificationsListFooter'),
        ListPlaceholder: () => import('@Core/components/List/ListPlaceholder'),
    },
    computed: {
        ...mapState('notifications', {
            notifications: state => state.notifications,
            notificationsCount: state => state.count,
        }),
        whiteColor() {
            return WHITE;
        },
    },
    methods: {
        ...mapActions('notifications', [
            'requestForNotifications',
        ]),
        navigateToAllNotifications() {
            this.$router.push({ name: 'notifications-grid' });
        },
        onFocus({ focus, isClickedInsideMenu }) {
            if (focus && !isClickedInsideMenu) {
                this.requestForNotifications();
            }
        },
    },
};
</script>

<style lang="scss" scoped>
    .notifications-dropdown {
        display: flex;
        flex: 1;
        flex-direction: column;
        width: 400px;
        height: calc(100vh - 48px);
    }
</style>
