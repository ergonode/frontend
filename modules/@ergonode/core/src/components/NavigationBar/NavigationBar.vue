/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <nav class="navigation-bar">
        <section class="navigation-bar__breadcrumbs">
            <slot name="breadcrumbs" />
        </section>
        <section class="navigation-bar__actions">
            <slot name="actions" />
        </section>
    </nav>
</template>

<script>
export default {
    name: 'NavigationBar',
};
</script>

<style lang="scss" scoped>
    .navigation-bar {
        display: flex;
        justify-content: space-between;
        height: 48px;
        background-color: $GRAPHITE_COAL;

        &__breadcrumbs {
            display: grid;
            grid-auto-flow: column;
            grid-column-gap: 8px;
            align-items: center;
            padding: 0 16px;
            box-sizing: border-box;
        }

        &__actions {
            display: grid;
            grid-auto-flow: column;
            grid-column-gap: 1px;
            background-color: $GRAPHITE_DARK;
            border-left: 1px solid $GRAPHITE_DARK;
        }
    }
</style>
