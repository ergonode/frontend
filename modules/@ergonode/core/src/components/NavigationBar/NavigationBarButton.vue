/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <button :class="['bar-button', {'bar-button--selected': isSelected}]">
        <slot />
    </button>
</template>

<script>
export default {
    name: 'NavigationBarButton',
    props: {
        isSelected: {
            type: Boolean,
            default: false,
        },
    },
};
</script>

<style lang="scss" scoped>
    .bar-button {
        position: relative;
        display: flex;
        justify-content: center;
        align-items: center;
        border: none;
        padding: 0 24px;
        background-color: $GRAPHITE_COAL;
        cursor: pointer;
        outline: none;

        &:hover, &--selected {
            background-color: $GRAPHITE_DARK;
        }
    }
</style>
