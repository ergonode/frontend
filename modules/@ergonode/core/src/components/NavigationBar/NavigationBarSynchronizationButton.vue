/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton>
        <IconProcessing :fill-color="whiteColor" />
    </NavigationBarButton>
</template>

<script>
import {
    WHITE,
} from '@Core/assets/scss/_js-variables/colors.scss';

export default {
    name: 'NavigationBarSynchronizationButton',
    components: {
        NavigationBarButton: () => import('@Core/components/NavigationBar/NavigationBarButton'),
        IconProcessing: () => import('@Core/components/Icons/Others/IconProcessing'),
    },
    computed: {
        whiteColor() {
            return WHITE;
        },
    },
};
</script>
