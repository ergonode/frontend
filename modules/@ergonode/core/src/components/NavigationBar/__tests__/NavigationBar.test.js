/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import NavigationBar from '@Core/components/NavigationBar/NavigationBar';
import {
    createLocalVue,
    shallowMount,
} from '@vue/test-utils';

const localVue = createLocalVue();
describe('NavigationBar', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(NavigationBar, {
            localVue,
        });
    });

    it('Component is named well', () => {
        expect(typeof NavigationBar.name).toBe('string');
        expect(NavigationBar.name).toEqual('NavigationBar');
    });

    it('Has no items', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
