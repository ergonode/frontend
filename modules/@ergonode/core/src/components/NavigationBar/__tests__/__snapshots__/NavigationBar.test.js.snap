// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Has no items 1`] = `
<nav class="navigation-bar">
  <section class="navigation-bar__breadcrumbs"></section>
  <section class="navigation-bar__actions"></section>
</nav>
`;
